Expose Channel Last 3d enum

glaringlee opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#32947 Expose Channel Last 3d enum**

This is the 1st step for supporting Channels Last 3d memory format.
Expose new enum, eliminated all compile warning caused by not handling the new enum inside switch case control flow.

Differential Revision: [D19707716](https://our.internmc.facebook.com/intern/diff/D19707716)

Commit:
Expose Channel Last 3d enum

[ghstack-poisoned]

kostmo commented:
## :pill: CircleCI build failures summary and remediations
As of commit cad01e35:


* **1/1** failures introduced in this PR


---

<details><summary>This comment was automatically generated by <a href="https://github.com/kostmo/circleci-failure-tracker/tree/master/docs/from-pull-request-comment">Dr. CI</a> (expand for details).</summary>Follow <a href="https://dr.pytorch.org/admin/comments-opt-out.html">this link to opt-out</a> of these comments for your Pull Requests.

Please report bugs/suggestions on the <a href="https://github.com/kostmo/circleci-failure-tracker/issues">GitHub issue tracker</a>.
</details>

VitalyFedyunin commented on c10/core/TensorImpl.h:
```diff
@@ -1356,6 +1356,13 @@ struct C10_API TensorImpl : public c10::intrusive_ptr_target {
+            dim() == 5,
+            "required rank 5 tensor to use channels_last_3d format");
+        //TODO Implement set_sizes_and_strides for channels last 3d
+        break;
```
Please trow unimplemented error here

Commit:
Update on "Expose Channel Last 3d enum"


This is the 1st step for supporting Channels Last 3d memory format.
Expose new enum, eliminated all compile warning caused by not handling the new enum inside switch case control flow.

Differential Revision: [D19707716](https://our.internmc.facebook.com/intern/diff/D19707716)

[ghstack-poisoned]
